### PR TITLE
Regression Test to check if Decks with wildcard characters in their name can be searched.

### LINKED ISSUE #7394: Regression Test: Search for decks with wildcard
It'd be nice to have a regression test to ensure that there's no regressions for this issue. 

_Originally posted by @david-allison-1 in https://github.com/ankidroid/Anki-Android/pull/7392#issuecomment-709631885_

### PR DESCRIPTION
**Pull request by anadi198**
## Purpose / Description
There was no test written to check if the Decks with wildcard characters in their names can be searched correctly(first doing an exact match search and if that fails, doing a regex based search). This PR adds a test inside FinderTest.java making sure that a card is visible regardless of the deck containing wildcard characters. 

tl;dr: A test written to make sure issue #6288 doesn't occur.

The issue mentioned above will be fixed by PR #7392 after which this test will stop failing. 

## Fixes
Fixes #7394 

## Approach

- Created a Deck with the name `"*Yr1::Med2::CAS4::F4: Renal::BRS (zanki)::HY"`
- Saved the `did` for the deck above and created a note inside the deck.
- Opened CardBrowser.java through RoboelectricTest and selected our deck from the drop down menu.
- Asserted that the number of cards in CardBrowser is equal to 1.

## How Has This Been Tested?

Ran the above test both with and without PR #7392, test passed with PR, failed without it.

## Learning (optional, can help others)
Anyone new to testing can refer to #7394 where there's a lot of discussion and I've noted down some things I learnt.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

### CODE REVIEW

**Comment by anadi198** [comment]
A `hacktoberfest-accepted` label would be much appreciated :)

**Comment by mikehardy** [comment]
More than happy to tag it up for hacktoberfest but it is currently unhappy in CI:

```
com.ichi2.libanki.FinderTest > test_deckNameContainingWildcardCanBeSearched FAILED
    java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at com.ichi2.libanki.FinderTest.test_deckNameContainingWildcardCanBeSearched(FinderTest.java:357)
```

🤔  is it working locally? If so it could be a case of needing to put some "robolectric pauses" in for the looper to let async things finish in the slightly slower CI environment?

**Comment by anadi198** [comment]
> com.ichi2.libanki.FinderTest > test_deckNameContainingWildcardCanBeSearched FAILED
>     java.lang.AssertionError: expected:<1> but was:<0>
>         at org.junit.Assert.fail(Assert.java:89)
>         at org.junit.Assert.failNotEquals(Assert.java:835)
>         at org.junit.Assert.assertEquals(Assert.java:647)
>         at org.junit.Assert.assertEquals(Assert.java:633)
>         at com.ichi2.libanki.FinderTest.test_deckNameContainingWildcardCanBeSearched(FinderTest.java:357)
> ```


@mikehardy I had predicted this failure before, since the PR for which this test is written hasn't been merged yet: 

> tl;dr: A test written to make sure issue #6288 doesn't occur.
> 
> The issue mentioned above will be fixed by PR #7392 after which this test will stop failing.

The test failing is actually a positive sign in this case I guess? Because the current master should be failing for this test. Sorry if it is causing confusion and if I shouldn't have opened this PR before that one was merged.

> 
> 🤔 is it working locally? If so it could be a case of needing to put some "robolectric pauses" in for the looper to let async things finish in the slightly slower CI environment?

But I'm ready to add more pauses if it required.

**Comment by mikehardy** [comment]
@anadi198 I pulled your branch after merging the preceding PR for it, rebased it to the new master (with your PR included) and pushed it back to your branch on github. CI should pass now then I'll merge this - thanks!

**Comment by mikehardy** [comment]
Hello! with the 2.14 release we started accepting donations with the stated goal of accepting money from users to support maintenance and development of AnkiDroid

This PR was included in 2.14, and if you'd like to submit an expense to share in a portion of our hopefully-sustainable monthly disbursement, please read here and do so in the next couple of days:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process

This is our first time handling donations and attempting to compensate anyone with these processes, so please be flexible and patient with us

Thank you for your contribution to AnkiDroid!
